gateway/rpc_storage_handler.go

```diff
@@ -817,13 +820,69 @@ func (gw *Gateway) getSessionAndCreate(keyName string, r *RPCStorageHandler, isH
 	}
 }
 
+// ProcessOauthClientsOps performs the appropiate action for the received clients
+// it can be any of the Create,Update and Delete operations
+func (gw *Gateway) ProcessOauthClientsOps(clients map[string]string) {
+	for v, action := range clients {
+		// value is: APIID.ClientID.OrgID
+		eventValues := strings.Split(v, ".")
+		apiId := eventValues[0]
+		oauthClientId := eventValues[1]
+		orgID := eventValues[2]
+
+		store, _, err := gw.GetStorageForApi(apiId)
+		if err != nil {
+			log.Error("Could not get oath storage for api")
+			return
+		}
+		switch action {
+		case OauthClientAdded:
+			// on add: pull from rpc and save it in local redis
+			client, err := store.GetClient(oauthClientId)
+			if err != nil {
+				log.WithError(err).Error("Could not retrieve new oauth client information")
```

Comment:
Should we `return` after this fails or the expected behavior is to continue?

Reply:
Thank you for pointing this out, indeed it should be a "continue" statement, so other events are processed as well. I just uploaded the change